WordPress/Classes/Models/Blog.m

```diff
@@ -292,9 +293,34 @@ - (NSString *)postFormatTextFromSlug:(NSString *)postFormatSlug
 // WP.COM private blog.
 - (BOOL)isPrivate
 {
-    return (self.isHostedAtWPcom && [[self getOptionValue:@"blog_public"] isEqualToString:@"-1"]);
+    return (self.isHostedAtWPcom && [self.privacy isEqualToNumber:@(BlogPrivacyPrivate)]);
```

Comment:
We are refreshing  it on the sync call for options, have a look here https://github.com/wordpress-mobile/WordPress-iOS/pull/4182/files#diff-aa8d7c1e04a521777ceca3c3d2b9e128R635